### ISSUE TITLE
Built-in attribute `project-version` not resolvable

### ISSUE THREAD

**Issue by bmuschko** [asciidoctor/asciidoctor-gradle-plugin#276]
The [built-in attribute `project-version`](https://asciidoctor.org/docs/asciidoctor-gradle-plugin/#options-attributes) is not replaced at runtime when used in Asciidoc file. This behavior seems to have changed with version 1.5.9 of the plugin. Version 1.5.8.1 still works. The generated HTML after 1.5.9 seems to just render `{project-version}` instead of the actual value.

**Comment by ysb33r** [comment]
Will have a look. Maybe it was lost when `projectdir` was lost too.

**Comment by mojavelinux** [comment]
I don't think it's necessarily correct to hard link revnumber and
project-version. revnumber is the version of the document, which may not be
the version of the software it's documenting. A convention, perhaps, but
they can be different.

**Comment by ysb33r** [comment]
I think it is a safe default for `revnumber`. It can always be overridden by the build script author to something else. The new patches will allow for that.

**Comment by bmuschko** [comment]
I'd be happy with _no_ default attributes set by the plugin. It just has to be documented properly.

**Comment by mojavelinux** [comment]
> I think it is a safe default for revnumber

It's a safe default as long as the document can override it. That would mean it needs to be set with a trailing `@`.

**Comment by ysb33r** [comment]
@mojavelinux do you mean 

```
attributes revnumber : '1.2.3@'
```

or

```
attributes 'revnumber@' : '1.2.3'
```

**Comment by mojavelinux** [comment]
Either one, but the first works in older versions of core.

**Comment by ysb33r** [comment]
> Either one, but the first works in older versions of core.

In that case I would prefer the second as we'll not support anything < asciidoctorj-1.6.0 in 2.0 of the plugin. The 2nd does look like a cleaner solution to me. Agree?

Also if we use `@` for any default attributes injected by the plugin, it can be clear that they are generated by the plugin rather than specified by the user.

**Comment by mojavelinux** [comment]
> In that case I would prefer the second as we'll not support anything < asciidoctorj-1.6.0 in 2.0 of the plugin. The 2nd does look like a cleaner solution to me. Agree?

:+1:

**Comment by ysb33r** [comment]
Fix scheduled in 1.5.10.
